src/Compilers/Core/Portable/Operations/ControlFlowGraph.cs

```diff
@@ -220,11 +230,22 @@ public ControlFlowGraph GetLocalFunctionControlFlowGraph(IMethodSymbol localFunc
                 throw new ArgumentNullException(nameof(localFunction));
             }
 
-            if (!_localFunctionsMap.TryGetValue(localFunction, out (ControlFlowRegion enclosing, ILocalFunctionOperation operation, int ordinal) info))
+            if (!TryGetLocalFunctionControlFlowGraph(localFunction, cancellationToken, out var controlFlowGraph))
             {
                 throw new ArgumentOutOfRangeException(nameof(localFunction));
             }
 
+            return controlFlowGraph;
+        }
+
+        internal bool TryGetLocalFunctionControlFlowGraph(IMethodSymbol localFunction, CancellationToken cancellationToken, out ControlFlowGraph controlFlowGraph)
```

Comment:
Am I missing something, or is `TryGetAnonymousFunctionControlFlowGraph` basically the same function, except over a different map? Consider refactoring with a helper that takes a ref to the correct graph.

Reply:
I am not sure I understand your suggestion. `TryGetLocalFunctionControlFlowGraph` is basically the just refactoring out the existing code from `GetLocalFunctionControlFlowGraph`, but does not throw any exception if it does not succeed. This function is used in `GetLocalFunctionControlFlowGraph` and newly added `GetLocalFunctionControlFlowGraphInScope`.

---
In reply to: [223788574](https://github.com/dotnet/roslyn/pull/30395#discussion_r223788574) [](ancestors = 223788574)

Reply:
`TryGetLocalFunctionControlFlowGraph` and `TryGetAnonymousFunctionControlFlowGraph` are pretty much clones of each other, except that they take different maps. The suggestion is to make one helper that does the actual work and takes a ref to the actual graph (with appropriate generics).

---
In reply to: [223794770](https://github.com/dotnet/roslyn/pull/30395#discussion_r223794770) [](ancestors = 223794770,223788574)

Reply:
I think the implementations are also slightly different. There is additional assert `Debug.Assert(localFunction == LocalFunctions[info.ordinal]);` and access to `LocalFunctions` here versus no assert and access to only `_anonymousFunctionsMap` below. I agree that we can still adjust these to use a single generic method, but I'd let @AlekseyTs know his preference as he initially wrote this code.

---
In reply to: [223804899](https://github.com/dotnet/roslyn/pull/30395#discussion_r223804899) [](ancestors = 223804899,223794770,223788574)

Reply:
Ok. Just a suggestion, not a blocker.

---
In reply to: [223811033](https://github.com/dotnet/roslyn/pull/30395#discussion_r223811033) [](ancestors = 223811033,223804899,223794770,223788574)